avanza/avanza.py

```diff
@@ -1469,7 +1469,7 @@ def get_inspiration_list(self, list_type: ListType):
             )
         )
 
-    def get_chart_data(self, order_book_id: str, period: TimePeriod):
+    def get_chart_data(self, order_book_id: str, period: TimePeriod, resolution: Resolution):
```

Comment:
This is a breaking change, I would add it as an optional parameter instead, defaulted to whatever the resolution when not sending a resolution is.

That way it's only a minor version bump instead of a major one.

```python
# Replace Resolution.HOUR with whatever the default resolution is
def get_chart_data(self, order_book_id: str, period: TimePeriod, resolution: Resolution = Resolution.HOUR):
```

Reply:
You're absolutely right, it is a breaking change. I didn't notice myself, thanks!

The problem is that the default resolution depends on period. Default resolution for five years is _week_ and one year is _day_.

Should we add an Undefined enum for Resolution that is default? That way it's possible to use different paths. Something like:

```
if resolution == Resolution.UNDEFINED:
    return self.__call(
        HttpMethod.GET,
        Route.CHARTDATA_PATH.value.format(
            order_book_id,
            period.value.lower()
        )
    )
return self.__call(
    HttpMethod.GET,
    Route.CHARTDATA_RES_PATH.value.format(
        order_book_id,
        period.value.lower(),
        resolution.value.lower()
    )
)
```

Reply:
Okay, in that case I would say it would be better to mark the parameter as optional and default it to None, instead of adding a magic value to the enum.

```python
from typing import Optional

def get_chart_data(self, order_book_id: str, period: TimePeriod, resolution: Optional[Resolution] = None):
```

I would also suggest removing both query parameters from the route as well (I should go through and do this everywhere) and then add them via the options parameter to __call, like this:

```python
def get_chart_data(self, order_book_id: str, period: TimePeriod, resolution: Optional[Resolution] = None):
  options = {
   'timePeriod': timePeriod.value.lower()
  } 
  
  if resolution != None:
    options['resolution'] = resolution.value.lower()
  
  return self.__call(
    HttpMethod.GET,
    Route.CHARTDATA_RES_PATH.value.format(order_book_id),
    options
  )
```
That way we can have a single return for the method, and just add the query params automatically if the are provided.

Reply:
That's perfect! Elegant!